packages/shortest/src/types/config.ts

```diff
@@ -1,32 +1,119 @@
 import { z } from "zod";
+import {
+  LLMSupportedModels,
+  LLMSupportedModelsType,
+  LLMSupportedProviders,
+  LLMSupportedProvidersType,
+} from "./ai";
+import { getDefaultProviderModel } from "@/ai/client-provider";
+import { MakeOptional } from "@/utils/types";
 
-export interface ShortestConfig {
-  headless: boolean;
-  baseUrl: string;
-  testPattern: string;
-  anthropicKey?: string;
-  mailosaur?: {
-    apiKey?: string;
-    serverId?: string;
-  };
-}
-
+/**
+ * Schema for Mailosaur configuration.
+ */
 const mailosaurSchema = z
   .object({
     apiKey: z.string(),
     serverId: z.string(),
   })
   .optional();
 
-export const configSchema = z
+export interface MailosaurConfig {
+  apiKey: string;
+  serverId: string;
+}
+
+/**
+ * Schema for the LLM (AI) configuration.
+ * If no model is provided, falls back to a default based on the provider.
+ */
+const llmSchema = z.object({
+  provider: z.literal(LLMSupportedProviders.ANTHROPIC),
```

Comment:
Can this be simplified to something like:


```suggestion
  provider: z.enum(AI_PROVIDERS).default("anthropic"),
```

where

```javascript
export const AI_PROVIDERS = ["anthropic"] as const;
export type AIProvider = (typeof AI_PROVIDERS)[number];
```

Reply:
I believe this approach oversimplifies the issue. Could you explain its benefits?

Right now, only the model field is optional—which makes sense because we don’t want users to worry about specifying a model. In contrast, the provider and `apiKey` fields in the configuration are mandatory

With your proposal, users would need to figure out—perhaps by reading the documentation—that the default provider is set to Anthropics. This could lead to errors if someone sets an OpenAI key in the environment but doesn’t provide the corresponding configuration

In my view, it’s much clearer when everything is explicitly defined in the configuration file. Ultimately, most users will end up setting the `ai.provider` field anyway, since only one default provider can be active even if many are available